apps/contact-service/src/routes/contacts/importContactsQuotaService.ts

```diff
@@ -0,0 +1,177 @@
+import {Schema} from '@effect/schema'
+import {UnexpectedServerError} from '@vexl-next/domain/src/general/commonErrors'
+import {type HashedPhoneNumber} from '@vexl-next/domain/src/general/HashedPhoneNumber.brand'
+import {getNextMidnightOnSelectedDate} from '@vexl-next/domain/src/utility/UnixMilliseconds.brand'
+import {
+  ImportContactsQuotaReachedError,
+  InitialImportContactsQuotaReachedError,
+} from '@vexl-next/rest-api/src/services/contact/contracts'
+import {RedisService} from '@vexl-next/server-utils/src/RedisService'
+import {Context, Effect, Layer, Option} from 'effect'
+import {DateTime} from 'luxon'
+import {
+  importContactsCountQuotaConfig,
+  importContactsResetAfterDaysQuotaConfig,
+  initialImportContactsCountQuotaConfig,
+} from '../../configs'
+import {UserDbService} from '../../db/UserDbService'
+
+export const ImportContactsQuotaRecord = Schema.Int.pipe(
+  Schema.greaterThanOrEqualTo(0)
+)
+
+export const createQuotaRecordKey = (
+  hashedPhoneNumber: HashedPhoneNumber
+): string => `importContactsQuota:${hashedPhoneNumber}`
+
+export interface ImportContactsQuotaOperations {
+  checkAndIncrementImportContactsQuota: (
+    hashedPhoneNumber: HashedPhoneNumber
+  ) => (
+    numberOfNewImportedContacts: number
+  ) => Effect.Effect<
+    void,
+    | UnexpectedServerError
+    | ImportContactsQuotaReachedError
+    | InitialImportContactsQuotaReachedError
+  >
+}
+
+export class ImportContactsQuotaService extends Context.Tag(
+  'ImportContactsQuotaService'
+)<ImportContactsQuotaService, ImportContactsQuotaOperations>() {
+  static readonly Live = Layer.effect(
+    ImportContactsQuotaService,
+    Effect.gen(function* (_) {
+      const redis = yield* _(RedisService)
+      const userDb = yield* _(UserDbService)
+
+      return {
+        checkAndIncrementImportContactsQuota:
+          (hashedPhoneNumber) => (numberOfNewImportedContacts) =>
+            Effect.gen(function* (_) {
+              const quotaRecordKey = createQuotaRecordKey(hashedPhoneNumber)
+              const user = yield* _(userDb.findUserByHash(hashedPhoneNumber))
```

Comment:
```suggestion
              const user = yield* _(userDb.findUserByHash(hashedPhoneNumber), Effect.flatten)
```

this will make user into type: `Effect.Effect<User,... | ElementNotFoundError, ...>`